src/Aggregation/DateHistogram.php

```diff
@@ -7,10 +7,70 @@
  *
  * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html
  */
-class DateHistogram extends Histogram
+class DateHistogram extends AbstractSimpleAggregation
 {
+    use Traits\KeyedTrait;
+    use Traits\MissingTrait;
+
     public const DEFAULT_TIMEZONE_VALUE = 'UTC';
 
+    /**
+     * @param string     $name     the name of this aggregation
+     * @param string     $field    the name of the field on which to perform the aggregation
+     * @param int|string $interval the interval by which documents will be bucketed
+     */
+    public function __construct(string $name, string $field, $interval = null)
+    {
+        parent::__construct($name, $field);
+        $this->setField($field);
+
+        if (null !== $interval) {
+            trigger_deprecation('ruflin/elastica', '7.1.0', 'Argument 3 passed to "%s()" is deprecated, use "setDateInterval()" or "setCalendarInterval()" instead. It will be removed in 8.0.', __METHOD__);
+
+            $this->setParam('interval', $interval);
+        }
+    }
+
+    /**
+     * Set the interval by which documents will be bucketed.
+     *
+     * @deprecated Deprecated since 7.1.0
+     *
+     * @param int|string $interval
+     *
+     * @return $this
+     */
+    public function setInterval($interval): self
+    {
+        trigger_deprecation('ruflin/elastica', '7.1.0', 'The "%s()" method is deprecated, use "setDateInterval()" or "setCalendarInterval()" instead. It will be removed in 8.0.', __METHOD__);
```

Comment:
I'm dubious about this deprecation, as `interval` parameter is still legit for ES version lower than `7.2`. It was only deprecated in 7.2 version but is still supported (until at least 8.0).

I think we should remove the deprecation here. But I would let the deprecation in the constructor so that users would be required to update their code to explicitly call `setInterval()`, `setFixedInterval()` or `setCalendarInterval()`.

Later when bumping minimum version of ES to a version later than 7.2 (probably when bumping to 8.0) we should be able to drop this method and only let the two others.

Would that make sense? ping @ruflin @thePanz

Reply:
There are two different deprecations:

* Deprecation in Elasticsearch
* Deprecation in Elastica

In the best case, they are aligned. As Elastica lags behind in releases, we kind of have the option to deprecate things already in older releases, assuming there is already an option available which is not deprecated.

To remove a method in 8.0, we should have it deprecated already in 7.x. I would say the earlier we can tell devs to adjust their code the better.

Reply:
Yes but that means in this case, that users of version older than 7.2 will get this deprecation but won't be able to remove the deprecation as the parameters (`setFixedInterval`, `setCalendarInterval`) can't work with their ES version. But maybe that's not an issue.

Reply:
I see. Maybe we can leave in a note in the docs for this. In general it would be great if we can push users to use the newest version of Elastica and the related version of Elasticsearch and would optimise for this. Maybe we could even adjust the README that this version of Elastica is supported for Elasticsearch >= 7.3 or similar. It does not mean, it will not work anymore with the older versions.

Reply:
I would add the PHP Doc to the methods, stating that those are not supported by ES >= 8.x
When the library will support ES v8.x only, we can remove those methods.
WDYT?

Reply:
No strong preference on my end. SGTM